File: 189-rotate-array/rotate-array.cpp
class Solution {
public:
   void rotate(vector<int>& nums, int k) {
    k = k % nums.size();
        int start = 0, end = nums.size()-1;
    while(start < end) {
        swap(nums[start],nums[end]);
        start++;
        end--;
    }
    start = 0, end = k - 1;
    while (start < end) {
        swap(nums[start],nums[end]);
        start++;
        end--;
    }
    start = k, end = nums.size()-1;
    while (start < end) {
        swap(nums[start],nums[end]);
        start++;
        end--;
    }
}
};

File: 189-rotate-array/rotate-array.py
class Solution(object):
    def rotate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        rotatedList = [0 for i in range(len(nums))]
        print(rotatedList)
        for i in range(len(nums)):
            newIndex = (i + k) % len(nums)
            print(newIndex)
            rotatedList[newIndex] = nums[i]
        nums[:] = rotatedList
        

